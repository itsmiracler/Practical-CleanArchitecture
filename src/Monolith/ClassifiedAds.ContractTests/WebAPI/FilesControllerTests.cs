using ApprovalTests;
using ApprovalTests.Reporters;
using ClassifiedAds.Application.AuditLogEntries.DTOs;
using ClassifiedAds.WebAPI.Controllers;
using ClassifiedAds.WebAPI.Models.Files;
using NJsonSchema;
using Xunit;

namespace ClassifiedAds.ContractTests.WebAPI
{
    [UseReporter(typeof(VisualStudioReporter))]
    public class FilesControllerTests
    {
        [Fact]
        public void FileEntryModelTest()
        {
            var schema = JsonSchema.FromType<FileEntryModel>().ToJson();
            Approvals.Verify(schema);
        }

        [Fact]
        public void UploadFileTest()
        {
            var schema = JsonSchema.FromType<UploadFile>().ToJson();
            Approvals.Verify(schema);
        }

        [Fact]
        public void AuditLogEntryDTOTest()
        {
            var schema = JsonSchema.FromType<AuditLogEntryDTO>().ToJson();
            Approvals.Verify(schema);
        }
    }
}
