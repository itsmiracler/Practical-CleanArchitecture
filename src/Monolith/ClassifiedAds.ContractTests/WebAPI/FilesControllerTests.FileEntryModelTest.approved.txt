{
  "$schema": "http://json-schema.org/draft-04/schema#",
  "title": "FileEntryModel",
  "type": "object",
  "additionalProperties": false,
  "properties": {
    "Id": {
      "type": "string",
      "format": "guid"
    },
    "Name": {
      "type": [
        "null",
        "string"
      ]
    },
    "Description": {
      "type": [
        "null",
        "string"
      ]
    },
    "Size": {
      "type": "integer",
      "format": "int64"
    },
    "UploadedTime": {
      "type": "string",
      "format": "date-time"
    },
    "FileName": {
      "type": [
        "null",
        "string"
      ]
    },
    "FileLocation": {
      "type": [
        "null",
        "string"
      ]
    },
    "Encrypted": {
      "type": "boolean"
    }
  }
}